Remove dependency of SILFunction in IRGenOptions

augusto2112 opened the pull request:
SILFunction::shouldBePreservedForDebugger checks if some optimizations are enabled to decide whether a function should be preserved so its accessible form the debugger or not. Some of these settings used to live only in IRGenOptions making SILFunction depend on IRGenOptions. This patch duplicates those settings into SILOptions to remove that dependency.

augusto2112 commented:
@swift-ci smoke test

slavapestov commented on lib/SIL/IR/SILFunction.cpp:
```diff
@@ -1002,12 +1001,11 @@ bool SILFunction::shouldBePreservedForDebugger() const {
-        options->LLVMLTOKind != IRGenLLVMLTOKind::None) {
-      return false;
-    }
+  const auto &silOptions = getModule().getOptions();
```
Why not just add a `bool ShouldBePreservedByDebugger` to SILOptions? Then you don’t have to much a bunch of stuff around.

augusto2112 committed:
Remove dependency of SILFunction in IRGenOptions

SILFunction::shouldBePreservedForDebugger checks if some optimizations
are enabled to decide whether a function should be preserved so its
accessible form the debugger or not. Some of these settings used to live
only in IRGenOptions making SILFunction depend on IRGenOptions.

augusto2112 commented:
@swift-ci smoke test

augusto2112 commented:
@swift-ci test

augusto2112 commented:
@swift-ci smoke test

slavapestov reviewed (commented):
Thanks!
